charts/agent/templates/_config.tpl

```diff
@@ -0,0 +1,27 @@
+{{- define "observe.deployment.applyClusterEventsConfig" -}}
+{{- $values := deepCopy .Values }}
+{{- $data := dict "Values" $values | mustMergeOverwrite (deepCopy .) }}
```

Comment:
I was misunderstanding how pipelines `|` work, this ends up taking the new dict `{ Values: (ouroldvalues) }` and using it as the last argument to mustMergeOverwrite (so the source that takes precedence). Think this is all in the name of using deepcopies for everything so that we dont accidentally change the global Values